### PR TITLE
Structured Logging migration: migrate package pkg/volume/util/operationexecutor for structured logging

### PR DESCRIPTION
**Pull request by sanwishe**
Structured Logging migration: migrate package pkg/volume/util/operationexecutor for structured logging

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Structured Logging migration: migrate package pkg/volume/util/operationexecutor
xref:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 1]
```diff
@@ -1516,7 +1508,7 @@ func (og *operationGenerator) GenerateExpandVolumeFunc(
 				return volumetypes.NewOperationContext(detailedErr, detailedErr, migrated)
 			}
 
-			klog.Infof("ExpandVolume succeeded for volume %s", util.GetPersistentVolumeClaimQualifiedName(pvc))
+			klog.InfoS("ExpandVolume succeeded for volume %s", util.GetPersistentVolumeClaimQualifiedName(pvc))
```
Comment (yangjunmyfm192085) [inline pkg/volume/util/operationexecutor/operation_generator.go:1511]:
```suggestion
			klog.InfoS("ExpandVolume succeeded for volume",  "volume", util.GetPersistentVolumeClaimQualifiedName(pvc))
```

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 2]
```diff
@@ -1528,15 +1520,15 @@ func (og *operationGenerator) GenerateExpandVolumeFunc(
 				return volumetypes.NewOperationContext(detailedErr, detailedErr, migrated)
 			}
 
-			klog.Infof("ExpandVolume.UpdatePV succeeded for volume %s", util.GetPersistentVolumeClaimQualifiedName(pvc))
+			klog.InfoS("ExpandVolume.UpdatePV succeeded for volume %s", util.GetPersistentVolumeClaimQualifiedName(pvc))
```
Comment (yangjunmyfm192085) [inline pkg/volume/util/operationexecutor/operation_generator.go:1523]:
```suggestion
			klog.InfoS("ExpandVolume.UpdatePV succeeded for volume",  "volume", util.GetPersistentVolumeClaimQualifiedName(pvc))
```

File: pkg/volume/util/operationexecutor/operation_executor.go [hunk 3]
```diff
@@ -707,28 +707,25 @@ func (oe *operationExecutor) VerifyVolumesAreAttached(
 		needIndividualVerifyVolumes := []AttachedVolume{}
 		for _, volumeAttached := range nodeAttachedVolumes {
 			if volumeAttached.VolumeSpec == nil {
-				klog.Errorf("VerifyVolumesAreAttached: nil spec for volume %s", volumeAttached.VolumeName)
+				klog.ErrorS(fmt.Errorf("VerifyVolumesAreAttached: nil spec for volume %s", volumeAttached.VolumeName), "")
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_executor.go:710]: Why are we not passing the volume name as an argument of `klog.ErrorS` directly?

File: pkg/volume/util/operationexecutor/operation_executor.go [hunk 4]
```diff
@@ -707,28 +707,25 @@ func (oe *operationExecutor) VerifyVolumesAreAttached(
...
 				oe.operationGenerator.GetVolumePluginMgr().FindPluginBySpec(volumeAttached.VolumeSpec)
 			if err != nil {
-				klog.Errorf(
-					"VolumesAreAttached.FindPluginBySpec failed for volume %q (spec.Name: %q) on node %q with error: %v",
-					volumeAttached.VolumeName,
-					volumeAttached.VolumeSpec.Name(),
-					volumeAttached.NodeName,
-					err)
+				klog.ErrorS(err, "VolumesAreAttached.FindPluginBySpec failed",
+					"volume", volumeAttached.VolumeName,
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_executor.go:718]:
```suggestion
					"volumeName", volumeAttached.VolumeName,
```

File: pkg/volume/util/operationexecutor/operation_executor.go [hunk 5]
```diff
@@ -707,28 +707,25 @@ func (oe *operationExecutor) VerifyVolumesAreAttached(
...
+					"spec.Name", volumeAttached.VolumeSpec.Name(),
+					"node", volumeAttached.NodeName)
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_executor.go:720]:
```suggestion
					"nodeName", volumeAttached.NodeName)
```

File: pkg/volume/util/operationexecutor/operation_executor.go [hunk 6]
```diff
@@ -707,28 +707,25 @@ func (oe *operationExecutor) VerifyVolumesAreAttached(
...
 				// should never happen since FindPluginBySpec always returns error if volumePlugin = nil
-				klog.Errorf(
-					"Failed to find volume plugin for volume %q (spec.Name: %q) on node %q",
-					volumeAttached.VolumeName,
-					volumeAttached.VolumeSpec.Name(),
-					volumeAttached.NodeName)
+				klog.ErrorS(nil, "Failed to find volume plugin for volume, volumePlugin is nil",
+					"volume", volumeAttached.VolumeName,
+					"spec.Name", volumeAttached.VolumeSpec.Name(),
+					"node", volumeAttached.NodeName)
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_executor.go:728]: Same as above

File: pkg/volume/util/operationexecutor/operation_executor.go [hunk 7]
```diff
@@ -763,7 +760,9 @@ func (oe *operationExecutor) VerifyVolumesAreAttached(
 		}
 		nodeError := oe.VerifyVolumesAreAttachedPerNode(needIndividualVerifyVolumes, node, actualStateOfWorld)
 		if nodeError != nil {
-			klog.Errorf("VerifyVolumesAreAttached failed for volumes %v, node %q with error %v", needIndividualVerifyVolumes, node, nodeError)
+			klog.ErrorS(nodeError, "VerifyVolumesAreAttached failed",
+				"volumes", needIndividualVerifyVolumes,
+				"node", node)
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_executor.go:765]: Is this the name?

File: pkg/volume/util/operationexecutor/operation_executor.go [hunk 8]
```diff
@@ -774,14 +773,16 @@ func (oe *operationExecutor) VerifyVolumesAreAttached(
 			volumeSpecMapByPlugin[pluginName],
 			actualStateOfWorld)
 		if err != nil {
-			klog.Errorf("BulkVerifyVolumes.GenerateBulkVolumeVerifyFunc error bulk verifying volumes for plugin %q with  %v", pluginName, err)
+			klog.ErrorS(err, "BulkVerifyVolumes.GenerateBulkVolumeVerifyFunc error bulk verifying volumes",
+				"volumePlugin", pluginName)
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_executor.go:777]:
```suggestion
				"volumePluginName", pluginName)
```

File: pkg/volume/util/operationexecutor/operation_executor.go [hunk 9]
```diff
@@ -774,14 +773,16 @@ func (oe *operationExecutor) VerifyVolumesAreAttached(
...
 		}
 
 		// Ugly hack to ensure - we don't do parallel bulk polling of same volume plugin
 		uniquePluginName := v1.UniqueVolumeName(pluginName)
 		err = oe.pendingOperations.Run(uniquePluginName, "" /* Pod Name */, "" /* nodeName */, generatedOperations)
 		if err != nil {
-			klog.Errorf("BulkVerifyVolumes.Run Error bulk volume verification for plugin %q  with %v", pluginName, err)
+			klog.ErrorS(err, "BulkVerifyVolumes.Run Error bulk volume verification",
+				"volumePlugin", pluginName)
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_executor.go:785]: Same as above

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 10]
```diff
@@ -166,13 +166,13 @@ func (og *operationGenerator) GenerateVolumesAreAttachedFunc(
 	// Iterate each volume spec and put them into a map index by the pluginName
 	for _, volumeAttached := range attachedVolumes {
 		if volumeAttached.VolumeSpec == nil {
-			klog.Errorf("VerifyVolumesAreAttached.GenerateVolumesAreAttachedFunc: nil spec for volume %s", volumeAttached.VolumeName)
+			klog.ErrorS(fmt.Errorf("VerifyVolumesAreAttached.GenerateVolumesAreAttachedFunc: nil spec for volume %s", volumeAttached.VolumeName), "")
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:169]: Same here - why are we not passing the argument directly?

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 11]
```diff
@@ -166,13 +166,13 @@ func (og *operationGenerator) GenerateVolumesAreAttachedFunc(
...
 			continue
 		}
 		volumePlugin, err :=
 			og.volumePluginMgr.FindPluginBySpec(volumeAttached.VolumeSpec)
 		if err != nil || volumePlugin == nil {
-			klog.Errorf(volumeAttached.GenerateErrorDetailed("VolumesAreAttached.FindPluginBySpec failed", err).Error())
+			klog.ErrorS(volumeAttached.GenerateErrorDetailed("VolumesAreAttached.FindPluginBySpec failed", err), "")
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:175]: Same question here

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 12]
```diff
@@ -193,36 +193,32 @@ func (og *operationGenerator) GenerateVolumesAreAttachedFunc(
 			attachableVolumePlugin, err :=
 				og.volumePluginMgr.FindAttachablePluginByName(pluginName)
 			if err != nil || attachableVolumePlugin == nil {
-				klog.Errorf(
-					"VolumeAreAttached.FindAttachablePluginBySpec failed for plugin %q with: %v",
-					pluginName,
-					err)
+				klog.ErrorS(err, "VolumeAreAttached.FindAttachablePluginBySpec failed",
+					"volumePlugin", pluginName)
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:197]:
```suggestion
					"volumePluginName", pluginName)
```

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 13]
```diff
@@ -193,36 +193,32 @@ func (og *operationGenerator) GenerateVolumesAreAttachedFunc(
...
 			}
 
 			volumeAttacher, newAttacherErr := attachableVolumePlugin.NewAttacher()
 			if newAttacherErr != nil {
-				klog.Errorf(
-					"VolumesAreAttached.NewAttacher failed for getting plugin %q with: %v",
-					pluginName,
-					newAttacherErr)
+				klog.ErrorS(newAttacherErr, "VolumesAreAttached.NewAttacher failed",
+					"volumePlugin", pluginName)
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:204]: Same as above

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 14]
```diff
@@ -193,36 +193,32 @@ func (og *operationGenerator) GenerateVolumesAreAttachedFunc(
...
 			}
 
 			attached, areAttachedErr := volumeAttacher.VolumesAreAttached(volumesSpecs, nodeName)
 			if areAttachedErr != nil {
-				klog.Errorf(
-					"VolumesAreAttached failed for checking on node %q with: %v",
-					nodeName,
-					areAttachedErr)
+				klog.ErrorS(areAttachedErr, "VolumesAreAttached failed",
+					"node", nodeName)
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:211]: Same as above

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 15]
```diff
@@ -193,36 +193,32 @@ func (og *operationGenerator) GenerateVolumesAreAttachedFunc(
...
 
 			for spec, check := range attached {
 				if !check {
 					actualStateOfWorld.MarkVolumeAsDetached(volumeSpecMap[spec], nodeName)
-					klog.V(1).Infof("VerifyVolumesAreAttached determined volume %q (spec.Name: %q) is no longer attached to node %q, therefore it was marked as detached.",
-						volumeSpecMap[spec], spec.Name(), nodeName)
+					klog.V(1).InfoS("VerifyVolumesAreAttached determined volume is no longer attached, therefore it was marked as detached.",
+						"volumeName", volumeSpecMap[spec],
+						"spec.Name", spec.Name(),
+						"node", nodeName)
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:221]: Same as above

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 16]
```diff
@@ -253,32 +249,28 @@ func (og *operationGenerator) GenerateBulkVolumeVerifyFunc(
 		attachableVolumePlugin, err :=
 			og.volumePluginMgr.FindAttachablePluginByName(pluginName)
 		if err != nil || attachableVolumePlugin == nil {
-			klog.Errorf(
-				"BulkVerifyVolume.FindAttachablePluginBySpec failed for plugin %q with: %v",
-				pluginName,
-				err)
+			klog.ErrorS(err, "BulkVerifyVolume.FindAttachablePluginBySpec failed",
+				"volumePlugin", pluginName)
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:253]: Same as above

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 17]
```diff
@@ -253,32 +249,28 @@ func (og *operationGenerator) GenerateBulkVolumeVerifyFunc(
...
 
 		volumeAttacher, newAttacherErr := attachableVolumePlugin.NewAttacher()
 
 		if newAttacherErr != nil {
-			klog.Errorf(
-				"BulkVerifyVolume.NewAttacher failed for getting plugin %q with: %v",
-				attachableVolumePlugin,
-				newAttacherErr)
+			klog.ErrorS(newAttacherErr, "BulkVerifyVolume.NewAttacher failed",
+				"volumePlugin", attachableVolumePlugin.GetPluginName())
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:261]: Same as above

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 18]
```diff
@@ -253,32 +249,28 @@ func (og *operationGenerator) GenerateBulkVolumeVerifyFunc(
...
 			return volumetypes.NewOperationContext(nil, nil, false)
 		}
 		bulkVolumeVerifier, ok := volumeAttacher.(volume.BulkVolumeVerifier)
 
 		if !ok {
-			klog.Errorf("BulkVerifyVolume failed to type assert attacher %q", bulkVolumeVerifier)
+			klog.ErrorS(fmt.Errorf("BulkVerifyVolume failed to type assert attacher %q", bulkVolumeVerifier), "")
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:267]: Same question as above here

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 19]
```diff
@@ -287,18 +279,18 @@ func (og *operationGenerator) GenerateBulkVolumeVerifyFunc(
 				nodeVolumeSpecs, nodeChecked := attached[nodeName]
 
 				if !nodeChecked {
-					klog.V(2).Infof("VerifyVolumesAreAttached.BulkVerifyVolumes failed for node %q and leaving volume %q as attached",
-						nodeName,
-						volumeSpec.Name())
+					klog.V(2).InfoS("VerifyVolumesAreAttached.BulkVerifyVolumes failed and leaving volume as attached",
+						"node", nodeName,
+						"volume", volumeSpec.Name())
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:284]: Same as above

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 20]
```diff
@@ -287,18 +279,18 @@ func (og *operationGenerator) GenerateBulkVolumeVerifyFunc(
...
 
 				check := nodeVolumeSpecs[volumeSpec]
 
 				if !check {
-					klog.V(2).Infof("VerifyVolumesAreAttached.BulkVerifyVolumes failed for node %q and volume %q",
-						nodeName,
-						volumeSpec.Name())
+					klog.V(2).InfoS("VerifyVolumesAreAttached.BulkVerifyVolumes failed",
+						"node", nodeName,
+						"volume", volumeSpec.Name())
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:293]: Same as above

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 21]
```diff
@@ -352,7 +344,8 @@ func (og *operationGenerator) GenerateAttachVolumeFunc(
 				volumeToAttach.VolumeSpec,
 				uncertainNode)
 			if addErr != nil {
-				klog.Errorf("AttachVolume.MarkVolumeAsUncertain fail to add the volume %q to actual state with %s", volumeToAttach.VolumeName, addErr)
+				klog.ErrorS(addErr, "AttachVolume.MarkVolumeAsUncertain fail to add the volume to actual state",
+					"volume", volumeToAttach.VolumeName)
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:348]: Here too

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 22]
```diff
@@ -1516,7 +1508,7 @@ func (og *operationGenerator) GenerateExpandVolumeFunc(
 				return volumetypes.NewOperationContext(detailedErr, detailedErr, migrated)
 			}
 
-			klog.Infof("ExpandVolume succeeded for volume %s", util.GetPersistentVolumeClaimQualifiedName(pvc))
+			klog.InfoS("ExpandVolume succeeded", "volume", util.GetPersistentVolumeClaimQualifiedName(pvc))
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:1511]: Same as above

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 23]
```diff
@@ -1528,15 +1520,15 @@ func (og *operationGenerator) GenerateExpandVolumeFunc(
 				return volumetypes.NewOperationContext(detailedErr, detailedErr, migrated)
 			}
 
-			klog.Infof("ExpandVolume.UpdatePV succeeded for volume %s", util.GetPersistentVolumeClaimQualifiedName(pvc))
+			klog.InfoS("ExpandVolume.UpdatePV succeeded", "volume", util.GetPersistentVolumeClaimQualifiedName(pvc))
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:1523]: Same as above

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 24]
```diff
@@ -1528,15 +1520,15 @@ func (og *operationGenerator) GenerateExpandVolumeFunc(
...
 		}
 
 		fsVolume, _ := util.CheckVolumeModeFilesystem(volumeSpec)
 		// No Cloudprovider resize needed, lets mark resizing as done
 		// Rest of the volume expand controller code will assume PVC as *not* resized until pvc.Status.Size
 		// reflects user requested size.
 		if !volumePlugin.RequiresFSResize() || !fsVolume {
-			klog.V(4).Infof("Controller resizing done for PVC %s", util.GetPersistentVolumeClaimQualifiedName(pvc))
+			klog.V(4).InfoS("Controller resizing done", "pvc", util.GetPersistentVolumeClaimQualifiedName(pvc))
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:1531]: Same as above

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 25]
```diff
@@ -1713,13 +1705,13 @@ func (og *operationGenerator) nodeExpandVolume(
 	actualStateOfWorld ActualStateOfWorldMounterUpdater,
 	rsOpts volume.NodeResizeOptions) (bool, error) {
 	if !utilfeature.DefaultFeatureGate.Enabled(features.ExpandPersistentVolumes) {
-		klog.V(4).Infof("Resizing is not enabled for this volume %s", volumeToMount.VolumeName)
+		klog.V(4).InfoS("Resizing is not enabled", "volume", volumeToMount.VolumeName)
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:1708]: same as above

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 26]
```diff
@@ -1822,7 +1814,7 @@ func isDeviceOpened(deviceToDetach AttachedVolume, hostUtil hostutil.HostUtils)
 		(devicePathErr != nil && strings.Contains(devicePathErr.Error(), "does not exist")) {
 		// not a device path or path doesn't exist
 		//TODO: refer to #36092
-		klog.V(3).Infof("The path isn't device path or doesn't exist. Skip checking device path: %s", deviceToDetach.DevicePath)
+		klog.V(3).InfoS("The path isn't device path or doesn't exist. Skip checking device", "path", deviceToDetach.DevicePath)
```
Comment (shubheksha) [inline pkg/volume/util/operationexecutor/operation_generator.go:1817]:
```suggestion
		klog.V(3).InfoS("The path isn't device path or doesn't exist. Skip checking device", "devicePath", deviceToDetach.DevicePath)
```

**Review by shubheksha — CHANGES_REQUESTED** [review]
Minor comments around clarity and consistency of key names and a question for a few error messages. Thank you for submitting this!

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 27]
```diff
@@ -927,7 +920,7 @@ func (og *operationGenerator) GenerateUnmountDeviceFunc(
 			return volumetypes.NewOperationContext(eventErr, detailedErr, migrated)
 		}
 
-		klog.Infof(deviceToDetach.GenerateMsg("UnmountDevice succeeded", ""))
+		klog.InfoS(deviceToDetach.GenerateMsg("UnmountDevice succeeded", ""))
```
Comment (njuptlzf) [inline pkg/volume/util/operationexecutor/operation_generator.go:923]:
```suggestion
		klog.InfoS(deviceToDetach.GenerateMsgDetailed("UnmountDevice succeeded", ""))
```

**Review by njuptlzf — CHANGES_REQUESTED** [review]
There should be `missing`?  `GenerateMsg` returns two parameters.

**Comment by shubheksha** [comment]
@sanwishe would you be able to take a look at this and rebase please?

**Comment by jingxu97** [comment]
@sanwishe sorry for the delay. Will you continue to work on this PR?

**Review by yangjunmyfm192085 — COMMENTED** [review]
/ping @sanwishe

**Review by yangjunmyfm192085 — COMMENTED** [review]
Let's rebase this PR, thanks.

**Comment by sanwishe** [comment]
> Let's rebase this PR, thanks.

sorry  for the  delay， i will rebase it.

**Review by yangjunmyfm192085 — COMMENTED** [review]
//remove-lifecycle rotten

**Review by yangjunmyfm192085 — COMMENTED** [review]
/lgtm

**Review by yangjunmyfm192085 — COMMENTED** [review]
/lgtm

**Review by yangjunmyfm192085 — COMMENTED** [review]
/cc @sanwishe

**Review by yangjunmyfm192085 — COMMENTED** [review]
Maybe we need rebase this pr.

**Pull request closed without merging**